docs/migration-service.md

```diff
@@ -122,96 +201,288 @@ flowchart TD
+}
 
-#### Cryptography
-A pair-wise key establishment scheme can be leveraged to establish a symmetric key. A suitable implementation can be found in [libsodium](https://doc.libsodium.org/) and its rust wrapper [libsodium-rs](https://docs.rs/crate/libsodium-rs/latest). Specifically, the **key exchange protocol based on X25519** would be a good fit for the given set-up (c.f. [libsodium](https://doc.libsodium.org/key_exchange) and [libsodium-rs](https://docs.rs/libsodium-rs/latest/libsodium_rs/crypto_kx/index.html)).
+/// Backup service authentication information
+pub struct BackupServiceInfo {
+    /// Ed25519 public key for mutual TLS authentication
+    pub public_key: Ed25519PublicKey,
+}
 
-The key establishment scheme requires that the node as well as the backup service have *mutually authenticated Curve25519 public keys*. The NEAR blockchain can be leveraged for this, more specifically, for each backup generation or recovery, the node and the backup service generate ephemeral keys on the `Curve25519` and publish them on the MPC smart contract. The node and the backup service can then each run a key generation protocol using their private key and the public key of the other party. 
+/// Destination node information during migration
+pub struct DestinationNodeInfo {
+    /// NEAR account public key (for verifying contract transaction signatures)
+    pub signer_account_pk: near_sdk::PublicKey,
 
-_Note: The curious reader might ask why this protocol does not simply use the NEAR public/private key pairs associated to the MPC node and the backup service. The reason for this is is twofold:_
-- _those keys are meant for signature generation._
-- _While it is true that Curve25519 used in X25519 and edwards25519 used by the NEAR blockchain are [birationally equivalent](https://crypto.stackexchange.com/questions/43013/what-does-birational-equivalence-mean-in-a-cryptographic-context), so one could theoretically convert the NEAR account keys and use them for `X25519`, it is generally advised to use one key per application. This also allows us to use ephemeral keys, as opposed to static keys for the encryption. Which is desirable._
+    /// New node's participant info (TLS key, cipher key, URL, etc.)
+    pub destination_node_info: ParticipantInfo,
+}
+```
+
+
+**Hard Launch Extensions:**
+
+For hard launch, the backup service must provide TEE attestation similar to MPC nodes. The contract will need to be extended to support attestation verification for backup services.
+
+**Planned Implementation Approach:**
+
+The backup service identification remains the same as soft launch: `AccountId` maps to `BackupServiceInfo`. For hard launch, the `BackupServiceInfo` struct would be extended to include attestation:
+
+```Rust
+/// Backup service authentication information (PLANNED extension for hard launch)
+/// Uses TeeState directly to store attestation and verification data
+pub type BackupServiceInfo = TeeState;
+```
```

Comment:
Huh, this is still the same issue, no?
`BackupServiceInfo` refers to a single backup service, but `TeeState` collects multiple attestations of different programs running inside TEEs.

I am not sure how we would extract the tls public key for a specific backup service from the `TeeState` and how we can map participant `AccountId`s to the corresponding backup service info with this design.